tests/Report/AbstractReporterTestCase.php

```diff
@@ -0,0 +1,203 @@
+                            'appliedFixers' => array('some_fixer_name_here'),
+                        ),
+                    ),
+                    0,
+                    0,
+                    false,
+                    false,
+                    false
+                ),
+            ),
+            'with diff' => array(
+                $this->createWithDiffReport(),
+                new ReportSummary(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here'),
+                            'diff' => 'this text is a diff ;)',
+                        ),
+                    ),
+                    0,
+                    0,
+                    false,
+                    false,
+                    false
+                ),
+            ),
+            'with applied fixers' => array(
+                $this->createWithAppliedFixersReport(),
+                new ReportSummary(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here_1', 'some_fixer_name_here_2'),
+                        ),
+                    ),
+                    0,
+                    0,
+                    true,
+                    false,
+                    false
+                ),
+            ),
+            'with time and memory' => array(
+                $this->createWithTimeAndMemoryReport(),
+                new ReportSummary(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here'),
+                        ),
+                    ),
+                    1234,
+                    2.5 * 1024 * 1024,
+                    false,
+                    false,
+                    false
+                ),
+            ),
+            'complex' => array(
+                $this->createComplexReport(),
+                new ReportSummary(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here_1', 'some_fixer_name_here_2'),
+                            'diff' => 'this text is a diff ;)',
+                        ),
+                        'anotherFile.php' => array(
+                            'appliedFixers' => array('another_fixer_name_here'),
+                            'diff' => 'another diff here ;)',
+                        ),
+                    ),
+                    1234,
+                    2.5 * 1024 * 1024,
+                    true,
+                    true,
+                    true
+                ),
+            ),
+        );
+    }
+
+    /**
+     * @return ReporterInterface
+     */
+    abstract protected function createReporter();
+
+    /**
+     * @return string
+     */
+    abstract protected function createExpectedFormat();
```

Comment:
The name is misleading IMO, why not simply `getFormat()`?

Reply:
done